### ISSUE TITLE
No detection when using HuggingFace models on GPU [BUG]

### ISSUE THREAD

**Issue by maxvelichko26** [voxel51/fiftyone#4555]
Hello and I want to say thank you for such a great product, though the following issue doesn't allow me to continue exploring it further. I'm trying to use a huggingface model (YoloForObjectDetection) with my .mp4 video. Everything works fine when using CPU, even though it takes a lot of time. But, when I try to use gpu via setting a torch.device, model doesn't perform any detection whatsoever and Fiftyone app only opens the original dataset video. During the process only one error is returned (it doesn't stop the app from opening): 
'Error: Input type (torch.FloatTensor) and weight type (torch.cuda.FloatTensor) should be the same or input should be a MKLDNN tensor and weight is a dense tensor'. 
If I understand correctly, this is due to model being on GPU and dataset being on CPU. This is probably a trivial problem, but I couldn't find any solution here or in other sources. Thank you for any help in advance!

Code below:

```
import fiftyone as fo
import torch
from fiftyone import ViewField as F
from transformers import YolosForObjectDetection, AutoImageProcessor
import fiftyone.zoo as foz

device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")


# I tried different approaches and reencoding, but the problem still persists
dataset= fo.Dataset.from_videos_patt("_____")
#dataset.name = "videos"
#dataset = fo.Dataset.from_videos(["_____"])
#dataset = fo.utils.video.reencode_videos(dataset, output_dir = '_______')
                 
model = YolosForObjectDetection.from_pretrained("hustvl/yolos-tiny")
model = model.to(device)

dataset.apply_model(model,label_field="yolovs", batch_size = 1)
#print(dataset)
session = fo.launch_app(dataset)

```

Tried on two Windows 10 machines, one with torch 2.3.1 + cuda 11.8, another with torch 2.2.0 and cuda 12.1.

output view usually looks like this:

```
Computing metadata...
 100% |█████████████████████| 1/1 [57.4ms elapsed, 0s remaining, 17.4 samples/s] 
Sample: 668d050db344b1c1208981e0
Error: Input type (torch.FloatTensor) and weight type (torch.cuda.FloatTensor) should be the same or input should be a MKLDNN tensor and weight is a dense tensor

 100% |█████████████████| 158/158 [274.0ms elapsed, 0s remaining, 583.6 samples/s] 

```

output doesn't have any labels or frames, just video. Tested on multiple examples
<img width="861" alt="image" src="https://github.com/voxel51/fiftyone/assets/115033570/30b60579-f2d8-4032-830d-7c2170854d3d">
